### ISSUE TITLE
Tweetinvi always showing null response Authentication time

### ISSUE THREAD

**Issue by SachinUnfyd** [linvi/tweetinvi#894]
Tweetinvi always showing null user.

 Auth.SetUserCredentials("lsiL1pwhHYc8CkPb","yMHcgeoNrnMGzXx2ixHoco8xq","965423104-dJwzaONw", "Cv6XBCth5Orn8DlKJCU9tf2sdmYzp0wHxQRD3GAKgx1qg");

var autuser = Tweetinvi.User.GetAuthenticatedUser();
var user = Tweetinvi.User.GetUserFromScreenName("@*********");

autuser or user get always null response

**Comment by SachinUnfyd** [comment]
This issue has been facing since last week.
Earlier, the same DLL used to have get data.

**Comment by p1pster** [comment]
@SachinUnfyd  did you resolve this as I'm currently getting exactly the same error. Constantly returning null.

**Comment by linvi** [comment]
@SachinUnfyd @p1pster did you both found a fix to the problem.

Were your credentials invalid?

**Comment by parseniev** [comment]
I am also facing this same issue - two MVC projects, for #1 everything works like a charm. For the other, #2 - authenticatedUser always comes back null as does basic Search.SearchTweets. 

The only difference I can think of is one project is using different core OWIN application authentication.. Using identical twitter credentials and api calls, #1 works a-ok and #2 always returns null - no exceptions, just null for search, authenticateduser, etc.

**Comment by linvi** [comment]
Could you share some code showing how you use the library please?

**Comment by iAutom8** [comment]
I have the same issue.  I created a new Windows Form Application using VS2015.   Installed TweetinviAPI  4.0.2 via NuGet.   Created button with following code (same as repository example) using valid Keys and Tokens:

```
Auth.SetUserCredentials("5xxxxxx",` "tdxxxxxxx" , "24823839-xxxxx", "Axxxxx");
var user = User.GetAuthenticatedUser();
```

`user` is NULL.  I even recreated the tokens and keys and still same issue.

I noticed the issue on Saturday with an app using Tweetinvi 0.9.10 which brought me here to try with the latest version.  I have also tried versions 3.0.0 and 4.0.1 without success

Thanks

**Comment by iAutom8** [comment]
**Update**
I ran the same test project using VS2019 and it worked!   Re-compiled my older project using 4.0.2 and appears to be working ok now.

**Comment by linvi** [comment]
Great happy to hear about it. You are the 2nd person telling me about this VS2015 having an issue, I will remember this :)

**Comment by linvi** [comment]
I do not provide support for this version sorry. I am spending already a lot of time maintaining and improving Tweetinvi. I cannot support such old versions. Either get the version on git, build it and test it out find a way to upgrade.

**Comment by parseniev** [comment]
@linvi I figured it out! Also just FYI this package was published June of this year, I'm not sure if it's yours directly or a fork off your project, but the github URL it lists is yours.

Reading up on the Exception Handling portion of your wiki, I set `ExceptionHandler.SwallowWebExceptions = false` and was able to see the exception that was being thrown:  `The underlying connection was closed: An unexpected error occurred on a send.`

I saw this same exception for an alternate twitter package I tried to switch to as a result of this issue, but when it came up I figured it was just out of date.

So, I read up on this issue [here](https://stackoverflow.com/questions/32788409/c-sharp-httpwebrequest-the-underlying-connection-was-closed-an-unexpected-error/32789483) and changing the security protocol to Tls1.2 fixed the issue. 

Above the request I just added
```
ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
```

I think that may be related to why switching to VS2019 for other people worked for this issue, but either way everything works, and it wasn't an issue with your code!

**Comment by linvi** [comment]
Great I am very happy to hear you found a solution. And thank you for reporting the solution. 

I would advise you to update maybe when version 5 comes out. Twitter api changes and various things have to change to catch up with them. 

Wish you the best in your project
